client/task-list/tasks/payments/index.js

```diff
@@ -73,21 +73,35 @@ class Payments extends Component {
 			: 'stripe';
 	}
 
-	markConfigured( method, queryParams = {} ) {
-		const { clearTaskStatusCache } = this.props;
+	async setMethodEnabledOption( method, value ) {
```

Comment:
this is the code that was persisting the enabled option extracted so it can be reused.

Reply:
Thanks, this is nice, although maybe we should add a check here to see if `enabled` is already set to the same `value`?
Several of the payment methods handle this in the component, so this currently would trigger duplicate update calls:
Stripe: https://github.com/woocommerce/woocommerce-admin/blob/main/client/task-list/tasks/payments/stripe.js#L137
Razorpay: https://github.com/woocommerce/woocommerce-admin/blob/main/client/task-list/tasks/payments/razorpay.js#L43

I believe Paypal as well, so we did just have to consolidate this.

Reply:
Yep great call, I resolved that here: https://github.com/woocommerce/woocommerce-admin/pull/6332/commits/f2c766263cc8e6a5dea502bb6449fdf99f4e6b65

also note I did a couple more things here:

1. I refactored the function to take a method name instead of relying on the method object, it was just much clearer to read.
2. I extracted the function so it could be tested and added some tests. I think this is simpler than trying to test it via the component itself. The component does way too much at the moment, would be great to break it down in future.